source/app.tsx

```diff
@@ -809,43 +808,48 @@ function AssistantMessageRenderer({ item }: {
       setThoughtsHeight(height);
     }
   }, [ thoughts ]);
+  
+  const showThoughts = thoughts && thoughts !== ""
 
 	return <Box>
     <Box marginRight={1} width={2} flexShrink={0} flexGrow={0}><Octo /></Box>
     <Box flexDirection="column" flexGrow={1}>
-      {
-        thoughts && thoughts !== "" && <Box flexDirection="column">
-          <Box
-            flexGrow={0}
-            flexShrink={1}
-            height={thoughtsOverflow > 0 ? MAX_THOUGHTBOX_HEIGHT : undefined}
-            width={MAX_THOUGHTBOX_WIDTH}
-            overflowY="hidden"
-            flexDirection="column"
-            borderColor="gray"
-            borderStyle="round"
-          >
-            <Box
-              ref={thoughtsRef}
-              flexShrink={0}
-              width={MAX_THOUGHTBOX_WIDTH - 2}
-              flexDirection="column"
-              marginTop={-1 * Math.max(0, thoughtsOverflow)}
-            >
+      {isStreamingContent ? (
+        <ScrollView height={scrollViewHeight}>
+          {
+            showThoughts && <Box flexDirection="column">
+              <Box
```

Comment:
also wouldn't be against removing the inner thoughtbox entirely, would get rid of some weird behavior where if you're not showing the full thoughtbox in the ScrollView it shows blank text
**When whole thoughtbox content is in view**
<img width="711" height="202" alt="Screenshot 2025-12-15 at 9 18 41 PM" src="https://github.com/user-attachments/assets/ff8e7cf0-b225-4e50-9db4-824e93abcccc" />
**When whole thoughtbox content is NOT in view**
<img width="710" height="230" alt="Screenshot 2025-12-15 at 9 18 27 PM" src="https://github.com/user-attachments/assets/38a47b35-8f82-4e89-916a-3bffd3d3d6e0" />